corehq/motech/value_source.py

```diff
@@ -274,21 +324,8 @@ def get_import_value(value_source, external_data):
 
 @singledispatch
 def get_commcare_value(value_source, case_trigger_info):
```

Comment:
Why can't this become a polymorphic method `<ValueSource>.get_commcare_value(case_trigger_info)`?

Reply:
It could. That's probably the most recognisable, and practical, approach.

`ValueSource` and its subclasses were intended to implement what *Clean Code* calls "Data Transfer Objects" (ch. 6), which sings their praises: "DTOs are very useful structures, especially when communicating with [say, remote APIs]. They often become the first in a series of translation stages that convert raw data ... into objects [like cases or tracked entities]."

... Except that's not really what they are. They could also be seen as "Data Classes", which *Refactoring* calls a "bad smell" (ch. 3). Martin Fowler agrees with you, and reckons I should move `get_commcare_value()` back.

If that's a good idea (and it probably is) I guess it makes sense to move `get_value()` and the rest of the functions back as well.

I'd have to open a new PR, cos then this one's name would be wrong. :upside_down_face: